compute/accelerator/benchmarks/mamf-finder.py

```diff
@@ -177,20 +184,40 @@ def benchmark_mm(m, n, k, dtype, device, num_iterations, num_warmup_iterations):
     start = arch.event(enable_timing=True)
     end = arch.event(enable_timing=True)
 
-    A = torch.randn(m, n, dtype=dtype, device=device)
-    B = torch.randn(n, k, dtype=dtype, device=device)
-    C = torch.empty(m, k, dtype=dtype, device=device)
-
-    times = np.zeros(num_iterations+num_warmup_iterations)
-    for i in range(num_warmup_iterations + num_iterations):
-        with torch.no_grad():
-            start.record()
-            torch.mm(A, B, out=C)
-            end.record()
-        arch.synchronize()
-        times[i] = start.elapsed_time(end)
+    if dtype == torch.float8_e4m3fn:
+        A = torch.randn(m, n, dtype=torch.float32, device=device).contiguous()
+        B = torch.randn(k, n, dtype=torch.float32, device=device).contiguous().t()
+        C = torch.empty(m, k, dtype=torch.float32, device=device)
+        
+        # 转换为 float8
+        A = A.to(torch.float8_e4m3fn)
+        B = B.to(torch.float8_e4m3fn)
+        C = C.to(torch.float8_e4m3fn)
+        
+        times = np.zeros(num_iterations+num_warmup_iterations)
+        for i in range(num_warmup_iterations + num_iterations):
+            with torch.no_grad():
+                start.record()
+                C = torch._scaled_mm(A, B)
+                end.record()
+            arch.synchronize()
+            times[i] = start.elapsed_time(end)
+    else:
+        A = torch.randn(m, n, dtype=dtype, device=device)
+        B = torch.randn(n, k, dtype=dtype, device=device)
+        C = torch.empty(m, k, dtype=dtype, device=device)
+        
+        times = np.zeros(num_iterations+num_warmup_iterations)
+        for i in range(num_warmup_iterations + num_iterations):
+            with torch.no_grad():
+                start.record()
+                torch.mm(A, B, out=C)
+                end.record()
+            arch.synchronize()
+            times[i] = start.elapsed_time(end)
```

Comment:
I thought of trying to avoid the main code block duplication in the proposed implementation and came up with this wrapper - not sure if it's actually any better readability-wise.

I guess it'd be useful should there will be more variations of the actual function being benchmarked.

Thoughts?
```
# Benchmark of a basic GEMM
def benchmark_mm(m, n, k, dtype, device, num_iterations, num_warmup_iterations):
    start = arch.event(enable_timing=True)
    end = arch.event(enable_timing=True)

    def time_it(iters=1):
        def decorator(func):
            def func_wrapper(*args, **kwargs):
                times = np.zeros(iters)
                for i in range(iters):
                    with torch.no_grad():
                        start.record()
                        ret = func(*args, **kwargs)
                        end.record()
                    arch.synchronize()
                    times[i] = start.elapsed_time(end)
                return times
            return func_wrapper
        return decorator

    total_iterations = num_iterations+num_warmup_iterations
    if dtype == torch.float8_e4m3fn:
        A = torch.randn(m, n, dtype=torch.float32, device=device).contiguous()
        B = torch.randn(k, n, dtype=torch.float32, device=device).contiguous().t()
        C = torch.empty(m, k, dtype=torch.float32, device=device)

        A = A.to(torch.float8_e4m3fn)
        B = B.to(torch.float8_e4m3fn)
        C = C.to(torch.float8_e4m3fn)

        @time_it(total_iterations)
        def time_iterations():
            C = torch._scaled_mm(A, B)

    else:
        A = torch.randn(m, n, dtype=dtype, device=device)
        B = torch.randn(n, k, dtype=dtype, device=device)
        C = torch.empty(m, k, dtype=dtype, device=device)

        @time_it(total_iterations)
        def time_iterations():
            torch.mm(A, B, out=C)

    times = time_iterations()[num_warmup_iterations:]
    elapsed_time = np.amin(times)/1000  # want the fastest
    tflops = (2 * m * n * k) / (elapsed_time * 10**12)
    return tflops
```

Reply:
I think it's great, let me apply these changes and test it out.